omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,341 @@
+	
+	@Before
+	public void init() throws Exception {
+		executeDataSet(RestTestConstants2_0.ALLERGY_TEST_DATA_XML);
+		executeDataSet(RestTestConstants2_0.OTHER_NON_CODED_CONCEPT_TEST_DATA_XML);
+		Allergen.setOtherNonCodedConceptUuid(RestTestConstants2_0.ALLERGY_OTHER_NON_CODED_UUID);
+	}
+	
+	@Override
+	public String getURI() {
+		return "patient/" + RestTestConstants2_0.PATIENT_UUID + "/allergy";
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_0.ALLERGY_UUID; // allergy does not support uuid
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return 4;
+	}
+	
+	/**
+	 *  Delete Allergy
+	 */
+	@Test
+	public void shouldDeleteUniqueAllergy() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Assert.assertFalse(allergy.isVoided());
+		
+		// attempt to delete allergy
+		handle(newDeleteRequest(getURI() + "/" + getUuid(), new Parameter("reason", "unit test")));
+		
+		allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+	}
+	
+	/**
+	 *  Delete Allergy incorrectly
+	 *  throw ObjectNotFoundException because allergy does not exist
+	 */
+	@Test(expected = ObjectNotFoundException.class)
+	public void shouldThrowExceptionWhenDeletingNonExistentAllergy() throws Exception
+	{
+		// attempt to delete allergy
+		handle(newDeleteRequest(getURI() + "/" + "NonExistentAllergyUUID", new Parameter("reason", "unit test")));
+	}
+
+	/**
+	 *  Delete all Allergies
+	 */
+	@Test
+	public void shouldDeleteAllAllergies() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		Allergies allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertEquals(Allergies.SEE_LIST, allergies.getAllergyStatus());
+		Assert.assertFalse(allergy.isVoided());
+
+		// attempt to delete all allergies
+		handle(newDeleteRequest(getURI(), new Parameter("reason", "unit test")));
+		
+		allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+		Assert.assertEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
+	}
+
+	/*
+	 *  Get Allergy 
+	 */
+	@Test
+	public void shouldGetAllergyByUuid() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+
+		// attempt to get allergy by uuid
+		SimpleObject savedAllergy = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+

```

Comment:
The example code does the same.How am I supposed to get the return Jason into a Java allergy object unless I use Jackson mapper?Are you saying I should use Jackson to deserialize but not to serialize?

Get Outlook for mobile




On Sat, Apr 9, 2016 at 11:24 AM -0700, "dkayiwa" <notifications@github.com> wrote:





> +             Assert.assertEquals("unit test", allergy.getVoidReason());
> +             Assert.assertEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
> +     }
> +
> +     /*
> +      *  Get Allergy
> +      */
> +     @Test
> +     public void shouldGetAllergyByUuid() throws Exception
> +     {
> +             Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
> +             Patient patient = allergy.getPatient();
> +
> +             // attempt to get allergy by uuid
> +             SimpleObject savedAllergy = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
> +

I thought we were replacing simple object by asserting the real returned allergy's expected property values.

---
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub:
https://github.com/openmrs/openmrs-module-webservices.rest/pull/195/files/88326ff12836bf08713921e6c12342f22e597ffb#r59119158